src/main/java/fr/ens/biologie/genomique/eoulsan/modules/diffana/DESeq2Checker.java

```diff
@@ -60,22 +89,77 @@ static boolean checkExperimentDesign(Experiment experiment, boolean throwsExcept
                     + experiment.getName(), throwsException);
         }
 
+        // Check if the column RepTechGroup is missing for the experiment
         if (!getExperimentSampleAllMetadataKeys(experiment)
                 .contains(REP_TECH_GROUP_KEY)
                 && !getAllSamplesMetadataKeys(experiment.getDesign()).contains(REP_TECH_GROUP_KEY)) {
             return error("RepTechGroup column missing for experiment: "
                     + experiment.getName(), throwsException);
         }
 
+        /* Check if there is no numeric character at the begin of a row in the column Condition
+        for a complex design model */
+        if (getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)) {
+            for (ExperimentSample es : experiment.getExperimentSamples()) {
+                String s = DesignUtils.getMetadata(es, CONDITION_KEY);
+                // Condition column contains an invalid numeric character as first character
+                if ( Character.isDigit(s.charAt(0)) && emd.getComparisons() != null){
+                    return error("One or more Condition rows start with a numeric character : "
+                            + experiment.getName(), throwsException);
+                }
+            }
+        }
+
+        // Check if there is no "-" in the column Condition when the contrast mode is activate
+        if (getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)) {
+            for (ExperimentSample es : experiment.getExperimentSamples()) {
+                String s = DesignUtils.getMetadata(es, CONDITION_KEY);
+                if (s.contains("-") && emd.isContrast()){
```

Comment:
`s.indexOf('-') != -1` is faster than a string comparison using `s.contains("-")`